pkg/operator/resource/resourceapply/core.go

```diff
@@ -3,42 +3,102 @@ package resourceapply
 import (
 	"bytes"
 	"context"
+	"crypto/md5"
 	"fmt"
+	"io"
 	"sort"
 	"strings"
 
-	"k8s.io/klog/v2"
-
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/equality"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/sets"
 	coreclientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/klog/v2"
 
 	"github.com/openshift/library-go/pkg/operator/events"
 	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
 )
 
+type CachedVersionKey struct {
+	name, kind, namespace string
+}
+
+type CachedResource struct {
+	resourceHash, resourceVersion string
+}
+
+// map of name,kind to resourceHash (of last applied resource) : resourceVersion(when resource was applied)
+var cachedResourceVersions map[CachedVersionKey]CachedResource = make(map[CachedVersionKey]CachedResource)
+
+func updateCachedResourceVersion(name string, kind string, namespace string, resourceHash string, resourceVersion string) {
```

Comment:
if this took a `runtime.Object` instead of a `resourceVersion`, you'd be able to simply call `updateCachedResourceVersion` and have it clear on nil (or do nothing) and avoid all the nil checks in callers.

Reply:
Done